deps/chakrashim/src/v8string.cc

```diff
@@ -91,8 +91,9 @@ int String::Write(uint16_t *buffer, int start, int length, int options) const {
   size_t count = 0;
   if (JsCopyStringUtf16((JsValueRef)this, start, length,
                          buffer, &count) == JsNoError) {
-    if (!(options & String::NO_NULL_TERMINATION)) {
-      buffer[count] = 0;
+    if (!(options & String::NO_NULL_TERMINATION) &&
+        (length == -1 || count < length)) {
```

Comment:
I only found this because of the VC debug heap asserts that hit when the null terminator was placed just after the end of the buffer (the incoming parameters left exactly enough room for the characters).  I checked in the V8 code and they do basically the exact same check, skipping the null terminator in that case.